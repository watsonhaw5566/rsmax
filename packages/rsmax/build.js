module.exports.buildApp = require('@rsmax/cli/lib/build').buildApp;
module.exports.buildMiniPlugin = require('@rsmax/cli/lib/build').buildMiniPlugin;
