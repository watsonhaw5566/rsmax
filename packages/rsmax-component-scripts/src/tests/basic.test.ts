import { buildComponent } from '../build';
import * as path from 'path';
import { Stats } from 'webpack';
import { join } from 'path';
import rimraf = require('rimraf');
import { readFileContent } from './helper/utils';
import * as fs from 'fs';

describe('test basic', () => {
  const dir = path.join(__dirname, './fixture/basic');

  beforeAll(() => {
    rimraf.sync(join(dir, 'dist'));
  });

  it('build esm', async () => {
    await buildComponent(dir, {
      type: 'esm',
      sourceDir: 'src',
      output: 'dist',
      babelrc: {
        plugins: [
          [
            'add-header-comment',
            {
              header: ['This string will be inserted in a comment header'],
            },
          ],
        ],
      },
    });

    ['dist/es/Foo/index.js', 'dist/es/Greet/index.js'].forEach(file => {
      const content = readFileContent(file, dir);
      expect(content).toMatchSnapshot();
    });
  });

  describe('build mini', function () {
    ['ali', 'wechat'].forEach((target: any) => {
      it('build ali', done => {
        const webpackCompiler = buildComponent(dir, {
          type: target,
          sourceDir: 'src',
          output: 'dist',
          mini: {
            input: {
              foo: 'Foo/index',
              greet: 'Greet/index',
            },
            configWebpack({ config }) {
              const rootNodeModules = path.join(__dirname, '../../../../node_modules');
              config.resolve.modules.add(path.join(__dirname, '../../../../node_modules'));
              config.resolve.modules.add(path.join(dir, './node_modules'));
              config.resolve.alias.clear();
              config.resolve.alias.set('react', path.join(rootNodeModules, 'react'));
            },
          },
          babelrc: {
            plugins: [
              [
                'add-header-comment',
                {
                  header: ['This string will be inserted in a comment header'],
                },
              ],
            ],
          },
        });

        webpackCompiler.hooks.done.tap('BuildStatsPlugin', (stats: Stats) => {
          const info = stats.toJson();
          if (stats && stats.hasErrors()) {
            info?.errors?.forEach(error => {
              console.error(error.message);
            });
            done(info.errors);
            return;
          }

          try {
            ['foo.js', 'foo.json', '__remax_runtime_options__.js', 'greet.js', 'greet.json'].forEach(file => {
              const f = path.join(dir, 'dist', target, file);
              fs.accessSync(f);
            });

            done();
          } catch (e) {
            done(e);
          }
        });
      });
    });
  });
});
