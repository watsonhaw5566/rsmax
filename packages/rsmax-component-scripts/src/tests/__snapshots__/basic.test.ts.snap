// Rstest Snapshot v1

exports[`test basic > build esm 1`] = `
"/**
* This string will be inserted in a comment header
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var React = require("react");
var one_1 = require("rsmax/one");
require("./index.less");
var Foo = function Foo(_a) {
  var size = _a.size,
    onClick = _a.onClick;
  return /*#__PURE__*/React.createElement(one_1.Button, {
    className: "button",
    onTap: onClick,
    style: {
      fontSize: size === 'default' ? 30 : 20
    }
  }, "text");
};
exports.default = Foo;"
`;

exports[`test basic > build esm 2`] = `
"/**
* This string will be inserted in a comment header
**/
import * as React from 'react';
import { View, Text, Image } from 'rsmax/one';
import styles from './index.scss';
export default (function (_ref) {
  var name = _ref.name;
  return /*#__PURE__*/React.createElement(View, {
    className: styles.greeting
  }, /*#__PURE__*/React.createElement(Text, null, "Hello111!!!!1 ", name), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Image, {
    mode: "widthFix",
    src: "https://gw.alicdn.com/tfs/TB1FvKecmR26e4jSZFEXXbwuXXa-140-140.svg"
  })));
});"
`;

exports[`test basic build esm 1`] = `
"/**
* This string will be inserted in a comment header
**/
\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var React = require(\\"react\\");

var one_1 = require(\\"rsmax/one\\");

require(\\"./index.less\\");

var Foo = function Foo(_a) {
  var size = _a.size,
      onClick = _a.onClick;
  return /*#__PURE__*/React.createElement(one_1.Button, {
    className: \\"button\\",
    onTap: onClick,
    style: {
      fontSize: size === 'default' ? 30 : 20
    }
  }, \\"text\\");
};

exports.default = Foo;"
`;

exports[`test basic build esm 2`] = `
"/**
* This string will be inserted in a comment header
**/

import * as React from 'react';
import { View, Text, Image } from 'rsmax/one';
import styles from './index.scss';
export default (function (_ref) {
  var name = _ref.name;
  return /*#__PURE__*/React.createElement(View, {
    className: styles.greeting
  }, /*#__PURE__*/React.createElement(Text, null, \\"Hello111!!!!1 \\", name), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Image, {
    mode: \\"widthFix\\",
    src: \\"https://gw.alicdn.com/tfs/TB1FvKecmR26e4jSZFEXXbwuXXa-140-140.svg\\"
  })));
});"
`;
